### PR TITLE
Add some tests for TreeGrid. Update README to reflect latest functionality.

### PR DESCRIPTION
**Pull request by alexstine**
## What?

The PR is adding some missing unit tests for the TreeGrid component. It is also clearing up the README.

## Why?

Chores are important to keep our code working. 👍 

## How?

Hopefully all checks will turn green and there will be more test data to prevent against regressions in the future.

## Testing Instructions

Wait for PR checks to turn green or test locally with the below command.

```
npm run test-unit -- packages/components/src/tree-grid/test/index.js
```

### CODE REVIEW

File: packages/components/src/tree-grid/index.js [hunk 1]
```diff
@@ -275,6 +275,10 @@ function TreeGrid(
 				);
 				focusablesInNextRow[ nextIndex ].focus();
 
+				// Let consumers know the row that was originally focused,
```
Comment (alexstine) [inline packages/components/src/tree-grid/index.js:278]: I guess this was missed in the original PR review. My bad.

File: packages/components/src/tree-grid/test/index.js [hunk 2]
```diff
@@ -97,16 +97,16 @@ describe( 'TreeGrid', () => {
 				</TreeGrid>
 			);
 
-			screen.getByText( 'Row 2' ).focus();
-			const row2Element = screen.getByText( 'Row 2' ).closest( 'tr' );
+			screen.getByText( 'Row 1' ).focus();
```
Comment (alexstine) [inline packages/components/src/tree-grid/test/index.js:100]: This was necessary as row 1 will do the collapsing/expanding now.

**Review by alexstine — COMMENTED** [review]
A few notes.

**Review by ciampo — APPROVED** [review]
Thank you as always, @alexstine!

Changes LGTM, although let's wait for @talldan 's review in case they have anything to add :)

One small note — could you just add an entry to the `@wordpress/components` CHANGELOG mentioning the change to the `onFocusRow` callback?

**Review by talldan — APPROVED** [review]
Looks good to me, too! Thanks for these.

**Comment by gziolo** [comment]
I cherry-picked this PR into Gutenberg 12.8 release to prevent issues with the merge conflicts in the CHANGELOG file while publishing to npm with the automated script.

**Pull request merged**